py/dict.go

```diff
@@ -189,6 +189,18 @@ func (d StringDict) M__getitem__(key Object) (Object, error) {
 	return nil, ExceptionNewf(KeyError, "%v", key)
 }
 
+func (d StringDict) M__delitem__(key Object) (Object, error) {
+	str, ok := key.(String)
+	if ok {
+		_, ok := d[string(str)]
+		if ok {
+			delete(d, string(str))
+			return None, nil
+		}
+	}
+	return nil, ExceptionNewf(KeyError, "%v", key)
```

Comment:
```suggestion
	str, ok := key.(String)
	if !ok {
		return nil, ExceptionNewf(KeyError, "%v", key)
	}
	_, ok = d[string(str)]
	if !ok {
		return nil, ExceptionNewf(KeyError, "%v", key)
	}
	delete(d, string(str))
	return None, nil
```

yes, it's longer, but only in the vertical dimension (however, it's less indented)